src/miniflask/util.py

```diff
@@ -2,20 +2,38 @@
 import argparse
 from argparse import Action
 from os import walk, path
+from pathlib import Path
 
 from colored import attr, fg
 
 
+# get total importable module name of directory, as seens from python
+# - i.e. $result can be loaded using `import $result`
+def get_full_base_module_name(directory):
```

Comment:
This looks like something `pkgutil` can probably help with? See e.g. https://docs.python.org/3/library/pkgutil.html#pkgutil.iter_modules

Reply:
Good idea to look into `pkgutil`.

Though, `iter_modules` does not fit the task, as it descends into all sub-directories.
The function we need iterates upwards in the directory structure to find the topmost module.

Reply:
Thanks for looking into this. Maybe there is a similar function in the pkgutil somewhere, I did not check for this.

Why do we need to traverse upwards? In the example above, we include all packages which are in the same directory (and downwards) where miniflask is used / init.

Reply:
The module directories could be placed anywhere in the project tree and do not need to be placed besides the file that actually loads these repositories.
Imagine some additional modules, say "utilities" are placed in `myproject/utilities`, and other modules, say "otherlibraries" are placed in `myproject/subpackage/otherlibraries`.

For each module repository, we need to know the path from `myproject` to the respective directory to know how to import these packages using pythons `import` statement.

Reply:
We should probably add a test-case for the described scenario.
I will push one later that reflects what I mean.

Reply:
I think we should avoid traversing paths (upwards) by ourself.
Instead, we should focus on finding all modules (that is, directories containing a `.module` file) anywhere in the available PATHs.

In the example above I think `myproject` would be in PATH, so all modules there can be found with `pkgutil`.
Otherwise, we end up in the same situation as before, where we rely on custom (and probably hacky ;-) ) functionality to import modules.

By using PATH together with `pkgutil` we have a reliable way to gather all available modules and can easily communicate this to the user, e.g. "every module in your PATH can be used".

Reply:
I very much like the idea.
I will look into this as soon as possible.

Reply:
One thing I am unsure about:
**If we traverse all available methods from `sys.path` do we actually want to include all compatible modules to the `modules_avail` list to be loaded?** In that case we would not need the `module_dirs`-argument in `mf.init` anymore.

**The alternative is to still enable the user to "prefilter" packages from `sys.path` that are of interest for a particular call of `mf.init`.** This can be done in several ways: inclusive (what packages to traverse into), exclusive (what packages to ignore), filter-function in some way (let the user do this).

**I suggest a "inclusive" traversal, where the user may specify a list of packages to traverse into.** If the user does not specify such a list all python packages are checked if they fit the requirements for a miniflask-module.

Reply:
I aggree that this is probably the best way forward!